src/app/domain/geolocation/helpers/quota.ts

```diff
@@ -0,0 +1,118 @@
+import { differenceInDays } from 'date-fns'
+
+import { Q } from 'cozy-client'
+import CozyClient from 'cozy-client'
+import flag from 'cozy-flags'
+import Minilog from 'cozy-minilog'
+
+import { showLocalNotification } from '/libs/notifications/notifications'
+import { t } from '/locales/i18n'
+import { getData, storeData, StorageKeys } from '/libs/localStore/storage'
+
+const MAX_DAYS_TO_CAPTURE_UNLIMITED = -1
+const ONE_DAY = 24 * 60 * 60 * 1000
+
+const log = Minilog('📍 Geolocation')
+
+export interface FirstTimeserie {
+  id: string
+  _id: string
+  type: string
+  startDate: string
+}
+
+const isMaxDaysToCaptureInvalid = (
+  maxDaysToCapture: number | null
+): maxDaysToCapture is null => {
+  return typeof maxDaysToCapture !== 'number'
+}
+
+const isMaxDaysToCaptureUnlimited = (maxDaysToCapture: number): boolean => {
+  return maxDaysToCapture === MAX_DAYS_TO_CAPTURE_UNLIMITED
+}
+
+const getFirstTimeserie = async (
+  client: CozyClient
+): Promise<FirstTimeserie | undefined> => {
+  try {
+    const firstTimeserieCachedLocally = await getData<FirstTimeserie>(
+      StorageKeys.FirstTimeserie
+    )
+
+    if (firstTimeserieCachedLocally) {
+      return firstTimeserieCachedLocally
+    } else {
+      const { data } = (await client.fetchQueryAndGetFromState({
```

Comment:
@paultranvan What do you think about this query ? 

Goal : first geojson timeserie ever